monitor-cockroachdb-with-prometheus.md

```diff
@@ -0,0 +1,154 @@
+
+<div id="toc"></div>
+
+## Before You Begin
+
+Make sure you have already started a CockroachDB cluster, either [locally](start-a-local-cluster.html) or in a [production environment](cloud-deployment.html). 
+
+## Step 1. Install Prometheus
+
+1. Download the [latest Prometheus tarball](https://prometheus.io/download/) for your OS.
+
+2. Extract the binary and add it to your `PATH`. This makes it easy to start Prometheus from any shell.
+
+3. Make sure Prometheus installed successfully:
+
+   ~~~ shell
+   $ prometheus -version
+   ~~~
+
+   ~~~
+   prometheus, version 1.4.1 (branch: master, revision: 2a89e8733f240d3cd57a6520b52c36ac4744ce12)
+     build user:       root@e685d23d8809
+     build date:       20161128-10:02:41
+     go version:       go1.7.3
+   ~~~
+
+## Step 2. Configure Prometheus
+
+1. Download the starter [Prometheus configuration file](https://github.com/cockroachdb/cockroach/blob/master/monitoring/prometheus.yml) and [aggregation rules](https://github.com/cockroachdb/cockroach/blob/master/monitoring/rules/aggregation.rules) for CockroachDB:
+   
+   ~~~ shell
+   # Configuration file:
+   $ wget https://raw.githubusercontent.com/cockroachdb/cockroach/master/monitoring/prometheus.yml \
+   -O prometheus.yml
+
+   # Aggregation rules:
+   $ wget -P rules https://raw.githubusercontent.com/cockroachdb/cockroach/master/monitoring/rules/aggregation.rules
+   ~~~
+
+   When you examine the configuration file, you'll see that it is set up to scrape the time series metrics of a single, insecure local node every 10 seconds:
+
+   - `scrape_interval: 10s` defines the scrape interval.
+   - `metrics_path: '/_status/vars'` defines the Prometheus-specific CockroachDB endpoint for scraping time series metrics.  
+   - `scheme: 'http'` specifies that the cluster being scraped is insecure.
+   - `targets: ['localhost:8080']` specifies the hostname and `http-port` of the CockroachDB node to collect time series metrics on. 
+
+2. Edit the configuration file to match your deployment scenario:
+
+   Scenario | Config Change
+   ---------|--------------
+   Multi-node local cluster | Expand the `targets` field to include `'localhost:<http-port>'` for each additional node.
+   Production cluster | Change the `targets` field to include `'<hostname>:<http-port>'` for each node in the cluster. Also, be sure your network configuration allows TCP communication on the specified ports.
+   Secure cluster | Uncomment `scheme: 'https'` and comment out `scheme: 'http'`.
+
+## Step 3. Start Prometheus
+
+1. Start Prometheus with the configuration file:
+
+   ~~~ shell
+   $ prometheus -config.file=prometheus.yml
```

Comment:
Yes, it seems to work fine. Is there something specific I should look for to indicate otherwise?